test/half.cpp

```diff
@@ -0,0 +1,1118 @@
+    {0xee59, -6500},
+    {0xee8a, -6696},
+    {0xee93, -6732},
+    {0xeed7, -7004},
+    {0xef0b, -7212},
+    {0xef59, -7524},
+    {0xef61, -7556},
+    {0xef67, -7580},
+    {0xefb6, -7896},
+    {0xf03a, -8656},
+    {0xf04e, -8816},
+    {0xf05f, -8952},
+    {0xf09f, -9464},
+    {0xf0c0, -9728},
+    {0xf173, -11160},
+    {0xf1d7, -11960},
+    {0xf225, -12584},
+    {0xf2ca, -13904},
+    {0xf2d8, -14016},
+    {0xf2e5, -14120},
+    {0xf317, -14520},
+    {0xf35d, -15080},
+    {0xf3bd, -15848},
+    {0xf3d3, -16024},
+    {0xf3e6, -16176},
+    {0xf3fb, -16344},
+    {0xf477, -18288},
+    {0xf4e0, -19968},
+    {0xf4e5, -20048},
+    {0xf50b, -20656},
+    {0xf5a2, -23072},
+    {0xf5c1, -23568},
+    {0xf634, -25408},
+    {0xf651, -25872},
+    {0xf68a, -26784},
+    {0xf69c, -27072},
+    {0xf6ce, -27872},
+    {0xf816, -33472},
+    {0xf849, -35104},
+    {0xf869, -36128},
+    {0xf878, -36608},
+    {0xf8cf, -39392},
+    {0xf90a, -41280},
+    {0xf916, -41664},
+    {0xf91e, -41920},
+    {0xf9c1, -47136},
+    {0xfa0a, -49472},
+    {0xfa11, -49696},
+    {0xfa1d, -50080},
+    {0xfa51, -51744},
+    {0xfa86, -53440},
+    {0xfaac, -54656},
+    {0xfb95, -62112},
+    {0xfbd1, -64032},
+    {0xfbe0, -64512},
+    {0xfbf5, -65184},
+    {0xfc00, -std::numeric_limits<float>::infinity()},
+    {0xfca5, std::numeric_limits<float>::quiet_NaN()},
+    {0xfcb9, std::numeric_limits<float>::quiet_NaN()},
+    {0xfcc6, std::numeric_limits<float>::quiet_NaN()},
+    {0xfd72, std::numeric_limits<float>::quiet_NaN()},
+    {0xfd77, std::numeric_limits<float>::quiet_NaN()},
+    {0xfda3, std::numeric_limits<float>::quiet_NaN()},
+    {0xfe3e, std::numeric_limits<float>::quiet_NaN()},
+    {0xfe89, std::numeric_limits<float>::quiet_NaN()},
+    {0xfe91, std::numeric_limits<float>::quiet_NaN()},
+    {0xfe93, std::numeric_limits<float>::quiet_NaN()},
+    {0xfed1, std::numeric_limits<float>::quiet_NaN()},
+    {0xff7a, std::numeric_limits<float>::quiet_NaN()},
+    {0xffa3, std::numeric_limits<float>::quiet_NaN()},
+};
+
+TEST_CASE(check_half_values)
+{
+    for(auto [x, f] : half_lut)
+    {
+        auto h = migraphx::bit_cast<migraphx::half>(x);
+        if(std::isnan(f))
+        {
+            CHECK(std::isnan(h));
+        }
+        else if(std::isinf(f))
+        {
+            CHECK(std::isinf(h));
+            CHECK((h < 0) == (f < 0));
+            CHECK(bit_equal(x, migraphx::half(f)));
+        }
+        else
+        {
+            CHECK(bit_equal(x, migraphx::half(f)));
+            CHECK(migraphx::float_equal(float(h), f));
+        }
+    }
+}
+
+TEST_CASE(check_flows)
+{
+    // check positive underflow
+    CHECK(bit_equal(std::numeric_limits<migraphx::half>::min() * std::numeric_limits<migraphx::half>::min(), migraphx::half(0)));
+
+    // check overflow
+    CHECK(bit_equal(std::numeric_limits<migraphx::half>::max() + std::numeric_limits<migraphx::half>::max(), std::numeric_limits<migraphx::half>::infinity()));
```

Comment:
You should also try max + min.

Reply:
max + min = max. I wonder if there is rounding happening.